src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/OutputWindowLogger.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.Logging;
+
+[Shared]
+[Export(typeof(OutputWindowLogger))]
+internal class OutputWindowLogger : ILogger
+{
+    private readonly OutputPane _outputPane;
+
+    [ImportingConstructor]
+    public OutputWindowLogger(JoinableTaskContext joinableTaskContext)
+    {
+        _outputPane = new OutputPane(joinableTaskContext);
+    }
+
+    [Obsolete("Exists only for Mock.")]
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    internal OutputWindowLogger()
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    {
+    }
+
+    public IDisposable BeginScope<TState>(TState state)
+    {
+        throw new NotImplementedException();
+    }
+
+    public bool IsEnabled(LogLevel logLevel)
+    {
+        return true;
+    }
+
+    public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
+    {
+        switch (logLevel)
+        {
+            case LogLevel.Critical:
+            case LogLevel.Error:
+            case LogLevel.Warning:
+                _outputPane.WriteLine(formatter(state, exception));
```

Comment:
I should probably change this so that the constructor takes a log level that is the min-bar.